### PR TITLE
feature/admin_api_audit_log

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Admin API Audit Log

Impacts
=======
pfperl after dispatching

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [x] Add acceptance tests (TestLink)

### CODE REVIEW

File: lib/pf/UnifiedApi/Controller/Config.pm [hunk 1]
```diff
@@ -1075,6 +1078,77 @@ sub fix_permissions {
...
+    my ($self) = @_;
+    my $stash = $self->stash;
+    my $status =  $self->res->code;
+    my $req = $self->req;
+    my $method = $req->method;
+    my $path = $req->url->path;
+    my $request = $self->make_audit_record_request();
+
+    my $current_user = $stash->{current_user};
+    return {
```
Comment (julsemaan) [inline lib/pf/UnifiedApi/Controller/Config.pm:1111]: You should also get the current tenant in this payload

File: lib/pf/UnifiedApi/Controller/Config/Bases.pm [hunk 2]
```diff
@@ -81,6 +81,14 @@ sub test_smtp {
     return $self->render(json => { message => 'Testing SMTP success' });
 }
 
+=head2 fields_to_mask
+
+fields_to_mask
+
+=cut
+
+sub fields_to_mask { qw(smtp_password pass galera_replication_password) }
```
Comment (julsemaan) [inline lib/pf/UnifiedApi/Controller/Config/Bases.pm:90]: You should also go through the fields of the scan engines and the firewall sso entries

File: lib/pf/UnifiedApi/Controller/Config.pm (hunk 1 above)
Comment (jrouzierinverse) [reply lib/pf/UnifiedApi/Controller/Config.pm:1111]: It is automatically added.

File: lib/pf/UnifiedApi/Controller/Config/Bases.pm (hunk 2 above)
Comment (jrouzierinverse) [reply lib/pf/UnifiedApi/Controller/Config/Bases.pm:90]: By default, any field named password will get masked.
Will verify if the scan engines have other fields that need to be masked.

Comment (jrouzierinverse) [reply lib/pf/UnifiedApi/Controller/Config/Bases.pm:90]: They are good.

**Comment by jrouzierinverse** [comment]
I reviewed and answered issues brought up by @julsemaan.

**Comment by extrafu** [comment]
@julsemaan Please update the NEWS file

**Comment by julsemaan** [comment]
@jrouzierinverse, can you provide the NEWS entry in your PR description next time so that the merge procedure handles that

**Pull request merged**